sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+review ACLs(Access Control Lists) should be instituted.
+
+Status: implemented / planned / to be implemented
+
+#### Core Infrastructure Initiative
+
+Cluster API asserts being 96% in progress with regards to
+the [Core Infrastructure Initiative (CII) badging process](https://bestpractices.coreinfrastructure.org/en/projects/5288)
+.
+
+##### DEV-02: Vulnerability report process is poorly defined
+
+The project has a SECURITY_CONTACTS file which nominally follows the Kubernetes
+vulnerability reporting process, but the file has not been maintained, and has
+not been invoked for subprojects as far as the authors are aware. In addition,
+the contacts only lists GitHub usernames - it isn’t clear how Kubernetes
+security is to get in touch with project maintainers in the case of private
+disclosure. To fix this, we would like to request a private google group, like
+this `cluster-api-security@kubernetes.io` that can be used for private
+discussions with SRC on any potential vulnerabilities discovered or reported for
+cluster-api.
+
+Status: [Planned](https://github.com/kubernetes-sigs/cluster-api/issues/5398)
+
+##### DEV-03: Project should configure some fuzzing
+
+Project lacked dynamic analysis / testing, to uncover reliability or security
+issues under stress testing.
+
+Status: [Planned](https://github.com/kubernetes-sigs/cluster-api/issues/6059)
+
+## Threat Modeling with STRIDE
+
+### Spoofing
+
+#### STRIDE-SPOOF-1 - Spoofing of Cloud Admin
+
+The identity of the Cloud Admin with access to Management Cluster can be spoofed
+by other users via stolen cloud and kubernetes credentials since cloud provider
+credentials are stored or are able to retrived from the workload cluster.
+
+For on-prem infrastructures, the Cloud Admin will have provisioned the cluster
+with wide ranging permissions, including the ability to create and delete VMs,
+add and remove disks etc... These by default will be written to the CPI
+credentials file on the disk of the workload control plane instances. A user of
+the cluster can start a pod with a host mount of /etc, and exfiltrate the cloud
+provider credentials.
+
+For hyperscaler environments like AWS, there is the ability to scope short lived
+credentials to the level of EC2 instances. However, the permissions are still
+broad for CPI and CSI to function, and a management cluster hosted in AWS will
+have even more permissions, such as the ability to create and destroy EC2
+instances. A user of the cluster can start a pod, and scrape the IMDS endpoint
+living on link-local (`http://169.254.169.254`) and retrieve these credentials.
+They are short-lived but a confused deputy attack is still possible.
+
+Cloud Admin has access to multiple Cloud services like EC2, ELB, Secrets
+Manager. Access to the management cluster basically provides this admin access
+to all the managed workload clusters and its workloads as well.
+
+#### STRIDE-SPOOF-1 Recommended mitigations
```

Comment:
I'm not really sure I understand the action required here.
Do we have corresponding issues? should they be created on Providers subprojects vs being in CAPI?

Reply:
I think the idea is that we need to add the recommendations that are not implemented in the list below into end user guidance page we will write in https://github.com/kubernetes-sigs/cluster-api/issues/6152 . Does that help resolve the confusion?